types/lodash/ts3.1/lodash-tests.ts

```diff
@@ -5201,6 +5201,7 @@ fp.now(); // $ExpectType number
     _.chain({ a: undefined }).get("a"); // $ExpectType never
     _.chain({ a: value }).get("a", defaultValue); // $ExpectType StringChain | PrimitiveChain<false> | PrimitiveChain<true>
     _.chain({ a: undefined }).get("a", defaultValue); // $ExpectType PrimitiveChain<false> | PrimitiveChain<true>
+    _.chain({ a: [1] }).get("a", []); // $ExpectType CollectionChain<number>
```

Comment:
If we support this use case for chaining, then we should also support it for the other variations of `get`, so add tests for those as well, e.g.

```ts
_.get({ a: [1] }, "a", []);
_({ a: [1] }).get("a", []);
```

Reply:
I'll add these tests, with the caveat that I have _not_ added this "workaround" to the types for `_.get()` and `_().get`.  Both of those signatures will continue to resolve to `number[] | never[]`.  

This PR doesn't actually change that behavior -- in my experience thus far, this appears to be far less of a problem in practice, as TypeScript appears to handle unions with `never[]` in a _mostly_-pragmatic fashion (ie. `.filter` predicates work like you'd expect them to, but `.map`'s argument gets widened to `any` for some reason)

If we to revisit this, I can add that to this PR (but I'd like to get some input from one of the maintainers first!). 

---

On the other hand, our original/underlying issue seems to be that `ExpChain<any[] | never[]>` expands to `CollectionChain<any> | CollectionChain<never>`, which is effectively an "impossible" type, as the predicates for many of the chained methods expand into something like `(val: any) => boolean | (val: never) => boolean` which can never be satisfied.

The tests don't really cover this scenario (ie. that one can actually chain multiple operations off of a `_.chain`), and I'm not really sure where to draw the line for providing test-coverage for this edge-case, as we're not going to reasonably be able to cover all possible combinations.  My test ensures that we don't end up with `CollectionChain<never>` in a union in this one narrow scenario, which seemed like a good-enough start.

---

Now that I've dug in a little further while writing this comment, I'm wondering if it makes more sense to remove my `_.chain().get()` workaround, and provide a more-explicit guards against ExpChain ever creating a `CollectionChain<never>`.

We'd change this:

```ts
type ExpChain<T> =
  // omitted a few lines...
  T extends List<infer U> | null | undefined ? CollectionChain<U> :
  // omitted a few lines...
  PrimitiveChain<T>;
```

to this:

```ts
type ExpChain<T> =
  // omitted a few lines...
  T extends List<infer U> | null | undefined ? U extends never ? never : CollectionChain<U> : 
  // omitted a few lines...
  PrimitiveChain<T>;
```

In some ways, this seems a bit better, but I'm not sure if it's only going to create _other_ undesirable scenarios.

Reply:
Makes sense. I don't think we want to always prevent CollectionChain<never>. Your current solution is probably best.